File: editorial_ch/chap3-3/prob3-3-3.py
# 輸入
n,r = map(int, input().split())

# 計算階乘
fact_n = 1
for i in range(1, n+1):
	fact_n *= i

fact_r = 1
for i in range(1, r+1):
	fact_r *= i

fact_nr = 1
for i in range(1, n-r+1):
	fact_nr *= i

# 輸出
print(int(fact_n / (fact_r * fact_nr)))

File: editorial_ch/chap3-3/prob3-3-3.cpp
#include <iostream>
using namespace std;

long long n, r;
long long Fact_n = 1;
long long Fact_r = 1;
long long Fact_nr = 1;

int main() {
	// 輸入
	cin >> n >> r;
	
	// 計算階乘
	for (int i = 1; i <= n; i++) Fact_n *= i;
	for (int i = 1; i <= r; i++) Fact_r *= i;
	for (int i = 1; i <= n - r; i++) Fact_nr *= i;
	
	// 輸出
	cout << Fact_n / (Fact_r * Fact_nr) << endl;
	return 0;
}